Fix part of #5914: UserSettingsModel and UserSubscriptionsModel Takeout

varun-tandon opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Implements takeout functionality on both UserSettingsModel and UserSubscriptionsModel as well as adds backend tests 

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [ ] The PR has an appropriate "CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [ ] The PR is assigned to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue.

ulloaluis committed:
Fix part of #5134, more robust testing to schema_utils.py

ulloaluis committed:
make requested changes

varun-tandon committed:
Added test for base model takeout

varun-tandon committed:
export data for UserSettingsModel and test for takeout

ulloaluis committed:
Takeout service. Export UserSubscriptionsModel

varun-tandon committed:
resolved linting

ulloaluis committed:
Merge branch 'takeout' of github.com:varun-tandon/oppia into takeout

seanlip commented on core/storage/base_model/gae_models_test.py:
```diff
@@ -43,6 +43,10 @@ def test_error_cases_for_get_method(self):
 
+    def test_export_data(self):
+        with self.assertRaises(NotImplementedError):
+            base_models.BaseModel.export_data('1')
```
Perhaps replace '1' with 'model_id' or similar -- more semantically understandable.

varun-tandon replied:
Done

varun-tandon replied:
Done

seanlip commented on core/storage/user/gae_models.py:
```diff
@@ -94,6 +94,18 @@ class UserSettingsModel(base_models.BaseModel):
 
+    @staticmethod
+    def export_data(user_id):
+        """(Takeout) Exports the data from UserSettingsModel into dictionary
```
Exports the data from UserSettingsModel into dict format for Takeout.

varun-tandon replied:
Done

seanlip commented on core/storage/user/gae_models.py:
```diff
@@ -94,6 +94,18 @@ class UserSettingsModel(base_models.BaseModel):
+        Returns:
+            dict. Dictionary of the data from UserSettingsModel.
+        """
+        return UserSettingsModel.get(user_id).__dict__
```
Nope, don't use `__dict__`. Avoid use of "internal methods" that have double underscores surrounding them. Compose the dictionary explicitly.

seanlip commented on core/storage/user/gae_models.py:
```diff
@@ -296,6 +308,23 @@ class UserSubscriptionsModel(base_models.BaseModel):
+        """Export UserSubscriptionsModel data."""
+        user_model = UserSubscriptionsModel.get(user_id, strict=False)
+
+        if not user_model:
```
if user_model is None

seanlip commented on core/storage/user/gae_models.py:
```diff
@@ -296,6 +308,23 @@ class UserSubscriptionsModel(base_models.BaseModel):
+        if not user_model:
+            user_model = UserSubscriptionsModel(id=user_id)
+
+        user_data = dict()
```
You can do:

    user_data = {
        'activities_ids': ...,
        ...
    }

Shouldn't it be activity_ids, btw?

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -44,6 +44,11 @@ def test_get_by_role(self):
+    def test_export_data(self):
+        user = user_models.UserSettingsModel.get_by_role(
+            feconf.ROLE_ID_ADMIN)[0]
+        self.assertEqual(user.__dict__, user.export_data(user.id))
```
Don't use `__dict__`. Manually construct the expected value for the test.

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -347,3 +352,16 @@ def test_get_categories_where_user_can_review(self):
+    USER_ID = 'user_id'
+
+    def test_export_data(self):
+        user_data = user_models.UserSubscriptionsModel.export_data(self.USER_ID)
```
This test does not take into account that there might be additional fields in user_data. Do an assertion on user_data itself rather than only on its fields.

Also, I suggest having a test where the values of the fields are non-trivial, to check format etc. Don't just aim for numerical test coverage; think about the behaviours you want to enforce and the things that might go wrong, and make sure that your tests cover these fully.

anubhavsinha98 commented:
Hi @varun-tandon, can you update your branch with develop. As we have now moved the lint tests to Travis CI. Thanks!

varun-tandon committed:
Merge branch 'develop' of https://github.com/oppia/oppia into takeout

ulloaluis committed:
Merge branch 'develop' of github.com:varun-tandon/oppia into takeout

merge

varun-tandon committed:
updated to individually export props

ulloaluis committed:
Made requested changes for UserSubscriptionsModel export_data.

DubeySandeep commented:
@varun-tandon, any update on this PR?

varun-tandon commented:
@DubeySandeep should be ready for second review within a few hours (currently fixing linting issues)

varun-tandon committed:
resolve linting problems

varun-tandon committed:
Merge branch 'takeout' of https://github.com/varun-tandon/oppia into takeout

varun-tandon committed:
resolve typo

seanlip commented on core/storage/user/gae_models.py:
```diff
@@ -94,6 +94,41 @@ class UserSettingsModel(base_models.BaseModel):
+            'username': user.username,
+            'normalized_username': user.normalized_username,
+            'last_agreed_to_terms': user.last_agreed_to_terms,
+            'last_started_state_editor_tutorial':
```
```
    'last_started_state_editor_tutorial': (
        user.last_started_state_editor_tutorial),
```
 
Similarly below, if you're wrapping across multiple lines.

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -44,6 +44,51 @@ def test_get_by_role(self):
+        user = user_models.UserSettingsModel.get_by_role(
+            feconf.ROLE_ID_ADMIN)[0]
+        user_data = user.export_data(user.id)
+        self.assertEqual(user.email, user_data['email'])
```
Use the actual field values rather than user.email, etc.

That will help us get some sense of any cases where we're only testing trivial things (e.g. if there are links to other model IDs in this model but we're only testing the empty-list case) and suggest areas where we should improve the test coverage.

varun-tandon replied:
@seanlip So to clarify, I have updated the code to testing None, default values, and empty lists, is this what we want?

seanlip replied:
Yes!

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -347,3 +392,48 @@ def test_get_categories_where_user_can_review(self):
+        user_models.UserSubscriptionsModel(id=self.USER_ID_1).put()
+
+        user_model_2 = user_models.UserSubscriptionsModel(id=self.USER_ID_2)
+        user_model_2.creator_ids.append(self.TEST_VALUE)
```
You should probably use different values for each of these ('sample_creator_id', 'sample_collection_id', etc.) to make sure that the right stuff is exported to the right fields.

varun-tandon replied:
@ulloaluis

ulloaluis replied:
Addressed and pushed to your takeout branch @varun-tandon 👍

varun-tandon committed:
Address issues with using user properties for assertions

varun-tandon committed:
resolve linting issues

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -44,6 +44,51 @@ def test_get_by_role(self):
+        self.assertEqual('user@example.com', user_data['email'])
+        self.assertEqual(feconf.ROLE_ID_ADMIN, user_data['role'])
+        self.assertEqual(None, user_data['username'])
+        self.assertEqual(
```
Here and below: don't use three lines where one or two will do.

seanlip commented on core/storage/user/gae_models_test.py:
```diff
@@ -44,6 +44,51 @@ def test_get_by_role(self):
             feconf.ROLE_ID_ADMIN)
         self.assertEqual(user[0].role, feconf.ROLE_ID_ADMIN)
 
+    def test_export_data(self):
```
We should also have a test for this where all field values are nontrivial (see my similar comment on the other model).

ulloaluis committed:
More robust testing

ulloaluis committed:
Merge branch 'takeout' of github.com:varun-tandon/oppia into takeout

updating my branch

ulloaluis committed:
small linting fix

ulloaluis committed:
linting error

ulloaluis committed:
linting error

ulloaluis committed:
linting error

ulloaluis committed:
linting error

varun-tandon committed:
Merge remote-tracking branch 'upstream/develop' into takeout

varun-tandon committed:
Merge branch 'takeout' of https://github.com/varun-tandon/oppia into takeout

seanlip committed:
Use pip_install command instead of vanilla pip install for PyGithub. (#6746)

aks681 committed:
Added an audit job to verify all images are present in the old and new directories (#6638)

* added image audit job

* made review changes

* minor change

* removed compiled_js

* fix lint

* minor change

* made review changes

* made review change

seanlip committed:
Routine update of translations. (#6747)

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Add Hungarian to list of supported languages.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix broken tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix translation error.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix translation error.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix errors in translations found by backend tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix broken tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix errors in translations.

* Fix a Python linting issue.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix spelling error in description

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix backend tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix French translation.

* Localisation updates from https://translatewiki.net.

* Fix French translation.

vojtechjelinek committed:
Speed webpack compilation (#6751)

* speed up webpack compilation

* compile webpack before running server

* add explaining comment

* add quicker compile to tests + formatting

* add typechecking

* dont specify files in tests

* add explaining comment

* use prod mode in backend test

* set DEV_MODE to true

* remove trailing space

ankita240796 committed:
Fixes creation of tmp folders while running lint (#6722)

jameesjohn committed:
Require collectionEditorStateService in collectionEditor (#6753)

Showtim3 committed:
fixed overflowing of the exploratory summary card (#6759)

nithusha21 committed:
Fix Array.fill() not a function stackdriver error using polyfill for IE (#6756)

* add polyfill for fill() function

* fix linting

Commit:
Fix part of #6550: Write backend tests for core/domain/param_domain.py and core/domain/search_services.py (#6664)

* Add initial tests

* Finish tests

* Fix lint errors

* Addressed commits

* Addressed first few comments

* Addressed comments

jameesjohn committed:
Fix part of #6565: Make story_editor, topic_editor and topic_viewer use pageTitleService (#6742)

* made topic editor to use pageTitleService

* made story editor to use pageTitleService

* made topic viewer use pageTitleService

* Update topic_editor.html

* Update topic_viewer.html

* made requested changes

* fix lint issues

* made necessary changes

* removed trailing comma

* made requested changes

* Made requested change

* remove unnecessary line

Commit:
Write tests for controllers/collection_editor (#6764)

aks681 committed:
Fixed ImagesAuditJob - Added image dimensions to names (#6763)

* fixed images job

* Fixed error

DubeySandeep commented:
@varun-tandon, any update on this PR?

ankita240796 committed:
Convert directives in extensions to component directives (#6752)

* Code repl

* Continue

* Drag and drop

* End exploration

* Fraction

* Graph Input

* Image click input

* Interactive map

* Item selection

* Logic proof

* Math

* Multiple choice

* Music notes

* Number with units

* Numeric input

* Pencil code editor

* Set Input

* Text Input

* Boolean editor

* Code string editor

* Coordtwodim editor

* Drag and drop html string editor

* Drag and drop int editor

* Filepath editor

* Fraction editor

* Graph editor

* Graph property editor

* Html editor

* Image with region editor

* Int editor

* List of set of html strings editor

* List of tabs editor

* List of unicode editor

* Logic error category editor

* Logic question editor

* Math latex editor

* Music phrase editor

* Non negative int editor

* Normalized string editor

* Number with units editor

* Parameter name editor

* Sanitized url and real editor

* Set of html string editor

* Set of unicode string editor

* Unicode string editor

* Collapsible

* Image

* Link

* Math

* Tabs

* Video

* Visualizations

* Fix test

* Fix comment

* Fix fraction input test

* Fix logic question test

* Fix graph input

* Fix multiple choice input test

* Fix number with units

* Fix number with units editor

seanlip committed:
Resolve npm errors (and fix some pre-push hook issues as well). (#6769)

* Fix NPM errors.

* Fix pre-push hook.

* Fix pre-push hook.

* Fix lint error.

Commit:
Fix part of #5134: Improve coverage for creator_dashboard and gae_search_services (#6480)

* Improve coverage

* Made tests robust

* Fix lint errors

* Address comments

* Address comments

* Address comments

* Addressed first few comments

* Finish addressing

* Addressed comments

* Addressed comments

* Addressed comments

* Divided test

* Removed mock clas

sstone-codaio committed:
Add a oneoff job to clean up invalid UserSubscriptionsModel (#6720)

* Add a oneoff job to clean up invalid UserSubscriptionsModel

* remove unused imported models

* Move cleanup logic out of ModelValidator class

* Address code review

* Fetch only exploration models from subscribers

* Added backend unit test

* Address code review

seanlip committed:
Add comment for npm dedupe command. (#6775)

DubeySandeep committed:
Translation tab text and speed improvements. (#6709)

* Translation tab text and speed improvments.

* Fixed incorrect tutorial url issue.

* Fix dependency issue.

vinitamurthi committed:
Fix part of #6346: Integrate Exploration player in Question player body (#6692)

* Draft PR that adds a question player directive

* Use the Question Player Backend API to fetch questions when starting the question player directive

* Fix linter error in StateTranslationEditorDirective.js

* Initial check in for question body

* Update question progression

* Stop getExplorationID from failing when ins question player mode

* Disable new structures players

* Fixed linter errors

* More linter fixes

* Fixed linter error

* Removed changes of empty lines

* Added exploration footer to the question player directive

* Fix linter errors

* More linter error fixes

* even more linter fixes

* Add question player backend dependency in exploration editor and player pages, this is to avoid an injector issue

* fix merge conflict in QuestionPlayerDirective

* Fixed merge conflict in state service

* Remove merge conflict messages

* Fix changes introduced in webpack

* Remove extra imports in exploration_editor.html

* More webpack fixes

* Remove print statement

* Removed extra imports

* Fixed linter issues

* Fix circle CI test failure

* Add topic name to practice sessions

DubeySandeep committed:
Fix #6686: Added a validation check for the patterns/file mentioned in the suites of e2e tests.  (#6716)

* Added e2e suites file/pattern check.

* Undo file rename changes.

ankita240796 committed:
Fix image upload issue (#6787)

DubeySandeep committed:
Update authors and changelog for v2.8.2. (#6784)

* Update authors and changelog for v2.8.2

* Minor grammatical changes in chnagelog.

aks681 committed:
Removed new structured editor flag (#6780)

YashJipkate committed:
Fix #6364: Added file overview check. (#6451)

* Added file overview check

* Addressed review

* Styling nits

* Added file overview part 1

* Added missing fileoverviews

* excluded conf files

* Added MANDATORY_PATTERNS_JS_REGEXP

* Refactored files

* Added copyright pattern

* Unified mandatory pattern checks for all types

* Address review

* Proper filename matching

* Modified MANDATORY_PATTERNS_REGEXP to pass tests

* Made into function

* Addressed review

* Fix lints

* Made checker function no-public

* Remove dist files

* Added missing file overviews

varun-tandon committed:
Resolved problems with code spanning lines

Commit:
Fix part of #6550: Improve coverage for learner dashboard (#6729)

* Tests for learner dashboard

* Addressed comments

* Add spaces

* Fix indent

apb7 committed:
Remove Browserstack from Travis (#6800)

NishealJ committed:
Add skip_files in third_party_size_check.py  (#6796)

* Update third_party_size_check.py

* Update code style and revert webpack version

* Remove print commands

* Remove un-necc comments

* Remove un-necc checks

* Revert package-lock.json

* Fix code style

* Fix spaces

* Fix Mistakes

* Fix spaces

* Fix spaces 2

* Fix Mistakes 2

* Add return-type

* Updates docstrings

vojtechjelinek committed:
Add bullet point about adding assignees (#6807)

* Add bullet point about adding assignees

* Edit based on review comments

bansalnitish committed:
Fix #6500: Removes console message if the querySegment length is longer (#6768)

* Remove console message

* Removed the corresponding test

* Added test

MLA98 committed:
Fix part of #6550: Improve test coverage of core/controller/libary.py" to 92% (#6760)

* test the scenerio that there is no top/ featured

* unittest for Index handler partially finished

* update library_test with test of top rated part

* CollectionSummariesHandlerTests empty case updated

* CollectionSummariesHandlerTests empty case updated

* CollectionSummariesHandlerTests empty case updated

* CollectionSummariesHandlerTests case that has collection updated

* CollectionSummariesHandlerTests linter fixed

* make it to 92%

* shorten the long lines and add some comments

* Addresses several review comments

* add explanation to language preference tests to review comment

* address some reviews

* address some reviews

* fix lint error

* remove blank lines

Commit:
Fix part of #6550: Improved test coverage for controllers/base.py (#6712)

amey-kudari committed:
Fixes #6331: Do not allow user to add roles if title is not given.  (#6625)

* Show warning if user tries to add roles to users without giving a title to the exploration, and write complete tests for this

* undo changes made in a different PR

* check for title warning, and fix the element not found error

* undo commits from other PR

* fix all nits and resolve merge conflicts

EricZLou committed:
Fix part of #6550: Increased coverage for learner progress services (#6643)

* 100% coverage for learner progress services

* Successful testing of Private. Incorrect function prototype.

* Helper functions for private activity

* Fix linting Errors

* New test structure

* Fix linting errors of import statements

* Add tests for re-publishing private exp/col

* Added some assert statements

* Some changes

* updated republishing an unpublished exp/col test

* add tests for exploration playlist and collection playlist

Commit:
Make coverage 100% for controllers/library (#6812)

seanlip committed:
Fix a bug where the email header was incorrect. (#6795)

* Fix erroneous email header.

* Add backend test.

jameesjohn committed:
Fix Part of #6565: Make Exploration pages and Collection pages use pageTitleService (#6786)

* pageTitleService in exploration pages

* make collection pages use pageTitleService

* fix e2e tests

* fix failing backend test

* address review comments

* work on failing backend tests

* addressed reviews

* address review comments

Showtim3 committed:
Extended hover on about menu when any li child is hovered (#6801)

* Extended hover on about menu when any li child is hovered

* updated with develop branch

* reverted package-lock.json

ankita240796 committed:
Migrate controllers/directives in the pages directory to component directives (#6749)

* Admin dev mode activities tab

* Admin prod mode activities tab

* Admin config tab

* Admin job tab

* Admin misc tab

* Role graph

* Admin role tab

* Admin navbar

* Collection editor tab

* Collection node creator

* Collection node editor

* Collection history tab

* Collection details editor

* Collection permission card

* Collection settings tab

* Collection statistics tab

* Collection editor navbar breadcrumb

* Collection editor navbar

* Collection footer

* Collection node list

* Activity tiles infinity grid

* Search bar

* Search results

* Question editor

* Question player

* Question list

* Update bindToController

* Update binding

* Fix collection editor

* Update controller naming

* Add admin directive

* Fix comment

* Add collection player directive

* Add collection local nav directive

* Add library directive

* Fix frontend and backend test

* Fix names

* Fix names

* Fix test

* Fix Admin test

* Fix search

* Fix watch

* Fix names

* Add file overview

* Fix names

* Fix test

seanlip committed:
Fix 500 error when downloading old exploration versions. (#6826)

* Fix 500 error when downloading old exploration versions.

* Remove width parameter.

* Address review comments.

NishealJ committed:
Upgrades uiSortable and jQuery  (#6803)

* Initial upgrade

* To trigger the build

Commit:
Fix part of #6550: Write backend tests for skill_jobs_one_off, story_jobs_one_off,  user_query_jobs_one_off , value_generators_domain, and visualization_registry (#6710)

* Added tests

* Addressed comments

* Add value generators test

* Fix backend tests

* Addressed comments

* Fix backend test

* Addressed comments

Commit:
Write tests for collection_domain (#6819)

mzaman07 committed:
Fix #5627 : Stop recording should stop microphone and angular-recorder should be replaced. (#6708)

sophiewu6 committed:
Fix #6548: Convert the current per-exploration translator role to a voice-artist role. (#6566)

mertdeg2 committed:
Fix part of #4057: Add unit tests for lookupEnvs method of ExpressionSyntaxTreeService.ts (#6828)

* Add unit tests for lookupEnvs method of ExpressionSyntaxTreeService.ts

This is part of the starter tasks: https://github.com/oppia/oppia/issues/4057

* Checkin

* Make sure help tooltip is visible

Fix for: https://github.com/oppia/oppia/issues/6262

* Revert last commit

sophiewu6 committed:
Fixed small bug in topic_viewer.html (#6838)

vibhor98 committed:
Fix #6806: Regressions in the Oppia navbars (#6820)

* Fi regressions in navbar

* Fix spacing within lists in RTE

* Remove unnecessary options in collection editor nav bar

* Made review changes.

Commit:
Fix part of #6550: Write tests for classifier, collection, email, feedback and user storage gae_models (#6802)

* Write tests

* Addressed comments

* Fix backend tests

* Addressed comments

* Addressed comments

vinitamurthi committed:
Score Ring component for the results page in practice sessions (#6813)

* Check in for score ring directive

* Review changes

* Fix review comments

apb7 committed:
Reduce the time for a PR to be classified as stale (#6837)

* Reduce the time for a PR to be classified as stale

* Modify the stale message

ankita240796 committed:
Fix issues with ctrl and (#6836)

anubhavsinha98 commented:
@varun-tandon, please update the PR? Thanks!

YashJipkate committed:
Convert standalone controllers in LibraryFooter and ExplorationGraph to component directives (#6840)

* Exploration Graph

* Library footer

* Oppia library footer

* Fix lint

* Fix CODEOWNERS

* Added missing EOL

Commit:
Fix part of #6550: Write tests for storage base_model, statistics and pages.py (#6811)

* Write tests

* Fix errors

* Make coverage 100%

* Addressed comments

* Change test names

* Addressed comments

* Address comments

* Fix lint

* Use args

Commit:
Fix part of #6550: Write tests for domain/collection_services (#6839)

* Write tests for domain/collection_services

* Write comments

vinitamurthi committed:
Fix a bug in the Question Player Directive (#6848)

* Fix questionplayerconfig bug in questionplayerdirective

* Remove new line in ConversationSkinDirective

jameesjohn committed:
Remove {{BEFORE_END_HEAD_TAG_HOOK}} from Base.html (#6835)

* remove before end head tag hook

* work on lint

* work on review changes and some refactoring

* work on failing test

* address review comments

* failing tests

* work on backend tests

* address review comments

seanlip committed:
Revert "Remove {{BEFORE_END_HEAD_TAG_HOOK}} from Base.html (#6835)" (#6852)

This reverts commit 56f26d2d20b26ef54929fbf3fabe01f20a48b0b0.

YashJipkate committed:
Convert standalone controllers in Exploration Editor page to component directives (#6843)

* Exploration editor tab

* Preview Tab

* Statistics Tab

* Settings Tab

* History Tab

* Fix karma of History Tab and Exploration Editor tab

* Feedback Tab

* Revert "History Tab"

This reverts commit 495007f627b2e2df38bee7211d3f1df67c242da3.

* Revert History tab spec changes

* Fix tests and restore HIstoryTabDirective

* Fix tests and lint

* Fix e2e test publicationAndLibrary

* Fix lint

* Fix lint

* Break line

* Maintained consistency

YashJipkate committed:
Convert Learner Local Nav and Leaner Local Info in ExplorationPlayer to component directives. (#6856)

* Learner Local Nav

* Learner View Info

* Learner View Info

* Controller -> Directive in fileoverview

YashJipkate committed:
Eliminated editr_tab.html and modified file overviews (#6858)

YashJipkate committed:
Introduce Folders by Feature to filters/ folder. (#6860)

* String utility filters

* Outside String utility filters

* Split tests

* Correct imports

* Fix lint

* Fix karma and relax naming

* Fix e2e

sstone-codaio committed:
Remove exploration from subscribers when it's deleted (#6719)

* Remove exploration from subscribers when it's deleted

* remove unused line

* Address comments and add unit test

* Address code review

* Fix unit test failure

* Add an option for user_jobs_continuous_test to disable deleting explorations from subscribers

* Fix unit test failures

* Address review comments

bansalnitish committed:
Fixes #6298: Adds a condition to check length of input response pairs (#6809)

* Add a check to ensure empty list is not accessed

* Fix comment

* Updated comment

* Fix failing tests

* Trailing spaces

aks681 committed:
Generalized image storage (#6783)

* Generalized image storage

* made review changes

* minor changes

* changed some comments

* lint

* made review changes

* Made review change

* added superclass

YashJipkate committed:
Apply folders by features to part of components/ folder (#6863)

* Button directives

* ck editor helpers

* common-layout-directives/common-elements

* common-layout-directives/navigation-bars

* entity-creation-services

* forms

* Relax lint

* Fix lint

Fix lint

* Fix CODEOWNERS

* Relax HTML naming

* Fix lint

* Fix lint

* Merge develop

* Fix backend test

* Fix backend for good

* Address comments

nithusha21 committed:
Fix #6737: Remove state id mapping from the codebase (#6830)

* remove all occurances of stateIdMappingModel

* linting fixes

* linting fixes

* add package-lock.json

* incorrect merge, fix package-lock file

* update jobs registry

* remove state id mapping domain class

* add back the model class

* add back tests for gae_models

* revert package-lock changes

* fix backend tests

apb7 commented:
Hi @varun-tandon, I am going ahead and closing this PR due to lack of response from your end. Please feel free to reopen it whenever you're working on it. Thanks!

Commit:
Fix part of #6550: Write tests for domain/classifier_domain and domain/event_services (#6831)

* add tests for event service

* Tests for classifier_domain

* Fix errors

* Address comments

* Use get

* Addressed comments

* Addressed comments

* Change parenthesis

* Removed extra parenthesis

* Move continue to top

YashJipkate committed:
Eliminate controller files for suggestion modals (#6859)

* Removed seperate controllers

* Fix lint

* Fix e2e

* Removed Show*

* Change names

* Fix lint

NishealJ committed:
Upgrades MathJax & MathJs (#6778)

* for travis

* for travis 2

* for travis 2

* for travis 2

* set circle ci

* set circle ci 2

* Rever travis and circle

* Delete upstream added for travis

* Initial upgrade

* initial upgrade

* Second upgrade

* Removed bin file

* Revert Travis

* Upgraded Maths-Expression

* Fix-mistakes

* update skip_files

* Remove un-used imports

* Change version in practise session

Commit:
Fix part of #6550: Write tests for gae_image_services and replaced PIL with Pillow (#6788)

* Write tests for gae_image_services and replaced PIL with Pillow

* Fix e2e

* Fix e2e

* Fix errors

* Fix selenium version

* Vendor add

* Add PIL to yamk

* Fix lint

* Use oppia_tools

* Addressed comments

* Fix lint

* Addressed comments

* Addressed comments

Commit:
Fix backend tests on develop (#6875)

YashJipkate committed:
Apply folders by features to remaining part of components/ folder. (#6869)

* Graph layout services

* Profile link directives

* Question directives

* Score ring

* ratings

* state-directives

* summary tile and version diff visualization

* state editor

* Fix lint

* Fix e2e

Commit:
Fix part of #6550: Write tests for collection_viewer and translator (#6741)

Commit:
Fix backend tests on develop (#6878)

YashJipkate committed:
Apply folders by feature to components/forms/validators/ and split tests. (#6870)

* Forms validators

* Split tests

brianrodri committed:
Fix #6523: Invalidate Stats Cache for States when Interaction Type Changes (#6679)

* delete stats cache for states whose interaction id changes

* fix conditional

* rename callback array

* use  interaction ids from backend

* only send necessary state data

* fix state collection logic

* pass interaction_ids in mocked out tests

* readability improvements

* remove debug statement

* avoid unnecessary refreshes

* s/Promise//g

* simplify setting exploration id

* use explicit naming

* condense overview

* move prefix to feconf

* target intended handler

* fix prefix usage

* add contextmanager for logging in as a user

* move login_session to TestBase

* remove login_session

* Revert package-lock.json changes

* add basic tests for StateAnswerStatisticsHandler

* Introduce login_context.

* use consistent fileoverviwew

* immediately resolve state deletion

* revert ESS spec

* use  instead of Promise

* revert package-lock.json

* return resolved promise

* Fix async tests.

* test multiple interactions

* add test for fetching answers

* fix no-break-after-hanging-indent

* reset unintentional pushes

* throw error when trying to update non-existing state

* just clear answers

* revert package-lock.json

* remove untested line

* revert package-lock.json

* More specific docstr.

* move login into try block

* move dangerous checks to initialize section

* login can't raise so revert it back

* fix stateStats reference

* minimize scope of interaction id fetching

* return empty viz info when interaction id is null

* be more forgiving with changes in interaction ids

* do not force unresolved answers to stay as the same array

* mock out complex callouts

* organization changes

* add test for null interaction ids

* Remove redundant comments

* check no requests are made

* use regex

* prefer O(1) look up

seanlip committed:
Modify update_configs.py script to include constants.js as well. (#6877)

* Modify update_configs.py script to include constants.js as well.

* Address review comments.

YashJipkate committed:
Apply folders by feature to lightweight pages. (#6876)

* About page, contact, donate

* Email dashboard

* Error pages, get started, landing pages

* Maintenance page

* Moderator page

* Notification dashboard page

* Practice session and preferences

* Privacy page

* Profile page

* Signup page

* Splash page

* Teach page

* Teach and terms page

* tests/ is now a sibling of pages/

* landing -> landing-pages

* Fix tests

* Fix tests

* Merge develop and address comment

Commit:
Write tests for domain/question_services (#6854)

seanlip committed:
Fix license link in README. (#6886)

vojtechjelinek committed:
Fix part of #6732: Migrate RTCs to webpack (#6882)

* migrate RTCs to webpack

* add overview

* modify tests webpack config

* add requires to RTCs

* remove html script-import file

* modify webpack configs

* remove file from CODEOWNERS

* Change test_html_contains_all_imports so that it looks in the ts file

* improve the backend test

jameesjohn committed:
Fix part of #5002: Remove isTopicManager and userIsLoggedIn (#6845)

YashJipkate committed:
Apply folders by feature to moderate sized pages. (#6879)

* Topics and skill dashboard

* Topic viewer

* Creator dashboard

* Library page

* Skill editor

* Modify CODEOWNERS

* Story editor

* Fix karma

* Fix e2e

* Collection player

* fix e2e

* Fix tests

* Fix e2e

Commit:
Fix part of #6550: Write tests for controllers/email_dashboard (#6744)

* Write tests for controllers/email_dashboard

* Addressed comments

* Addressed comments

* Addressed comments

* Addressed comments

* Remove parenthesis

* addressed comments

DubeySandeep committed:
Fix #6794: Add landing page for negative numbers and refactor landing page. (#6846)

* Added landing page for negative numbers and refactored landing page.

* Add css property in sorted order.

* Undo changes in packe-lock.json

* Add hypenstospaces filter and fix typos.

* Change landing_page_data structure to include topic_title.

* Added proper filename.

YashJipkate committed:
Apply folders by feature to Exploration Editor. (#6899)

* exploration_editor/ -> exploration-editor-page/

* Prepare subdirectories

* editor navigation

* editor tab

* exploration-objective-editor

* save and publish buttons

* title editor

* feedback tab

* history tab

* improvements tab

* modal templates

* param changes editor

* preview tab

* services

* settings tab

* statistics tab

* suggestion modal

* translation modal and main controller

* Fix lint and codeowners

* Fix tests

nithusha21 committed:
Fix #6897: fix issues with tile backgrounds and images (#6872)

nithusha21 committed:
One off job to delete state ID mapping models from the datastore. (#6871)

jameesjohn committed:
Serve get_started page statically (#6818)

* create transclusion components and work on serving get_started statically

* remove unnecessary require

* recommended changes

* refactor and made some changes to base.html

* lint issues

* requested fixes and refactoring

* lint issues

* fix webpack config

* work on failing tests

* review changes and test issues

* minor fix

* lint issues

* address review comments

* more work on frontend tests

* adress reviews

* address review comments

* address new review comments

* e2e tests for meta tags

* test fixes

* still on the tests

* some tests issues

* address review comments

* address review comments

* address review comments

* resolve issues with merge

* changed to controller binding

* use webpack require in maintenance and build_sources

* work on failing tests

* revert base in build_sources

* fix failing tests

* remove global

* address review comments

Commit:
Fix part of #6550: Write tests for domain/user_jobs_one_off (#6842)

* Write tests for domain/user_jobs_one_off

* Addressed comments

vojtechjelinek committed:
Fix part of #6732: Migrate value_generators to webpack (#6900)

* migrate value_generators to webpack

* fix backend tests

YashJipkate committed:
Apply folders by feature to Learner Dashboard.  (#6905)

Commit:
Fix part of #6550: Write tests for domain/topic_domain (#6873)

* Write tests for domain/topic_domain

* Test migrations

* Addressed comments

NishealJ committed:
Upgrades bleach and simplejson  (#6797)

vojtechjelinek committed:
Fix part of #6732: Migrate visualizations to webpack (#6885)

* migrate visualizations to webpack

* fix linting

* fix linting

* fix linting

* test coverage

* fix tests

jameesjohn committed:
Remove Before End Hook from base.html (#6880)

* remove before end head tag hook

* work on lint

* work on review changes and some refactoring

* work on failing test

* address review comments

* failing tests

* work on backend tests

* address review comments

* rework before end head hook

* refactor to use IIFE

ankita240796 committed:
Adds a check to track extra js files and removes local compiled js before each compilation (#6694)

YashJipkate committed:
Apply folders by feature to Exploration Player and Admin page (#6892)

* Exploration player

* Fix lint

* Fix e2e

* Admin page

* Fix e2e

* fix e2e

* Fix e2e

kevinlee12 committed:
Fix part of #6732: Extensions objects webpack integration (#6889)

* Add extensions folder to webpack config

* Convert extension objects dependencies to webpack

* Change extension objects import method for exploration editor

* Employ webpack for extension objects for exploration player

* Employ webpack for extension objects for practice sessions

* Employ webpack for extension objects for skill editor

* Employ webpack for extension objects for story editor

* Employ webpack for extension objects for topic editor

* Fix lint errors

* Add extensions to karma config

* Convert individual imports to single large import

* Convert individual imports to single large for for players

* Replace missing import for exploration editor

Commit:
Fix part of #6550:  Write tests for controllers/acl_decorators (#6782)

bansalnitish committed:
Add log for null summaries (issue #6157 ) (#6910)

* Add log for null summaries

* Address review comments

YashJipkate committed:
Separate constants into separate file per page (#6906)

* Admin page

* Collection player page

* Creator dashboard page

* Exploration editor page

* Exploration player page and (surprisingly) exploration editor leftovers

* Topic landing page and learner dashboard

* Library page

* Practice session page

* Skill editor

* Story editor

* Topics and skills dasboard

* Fix lint

YashJipkate committed:
Apply folders by feature to collection editor and topic editor. (#6890)

* collection editor

* Topic editor

* Fix lint and CODEOWNERS

* Merge develop proper

* Move HTML files

* Add the new files to CODEOWNERS

jameesjohn committed:
Milestone 1.2 Serve more pages statically (#6874)

* create transclusion components and work on serving get_started statically

* remove unnecessary require

* recommended changes

* refactor and made some changes to base.html

* lint issues

* requested fixes and refactoring

* lint issues

* fix webpack config

* work on failing tests

* review changes and test issues

* minor fix

* lint issues

* address review comments

* more work on frontend tests

* adress reviews

* address review comments

* address new review comments

* serve splash pages statically

* serve privacy page statically

* e2e tests for meta tags

* test fixes

* still on the tests

* some tests issues

* address review comments

* address review comments

* address review comments

* resolve issues with merge

* serve terms page statically

* serve teach page statically

* serve about page statically

* changed to controller binding

* use webpack require in maintenance and build_sources

* remove globals

* remove unnecessary line

* address review comments and work on failing tests

* remove head hook from get-started

* fix lint issue

* address review comments

* work on failing tests

* still on tests

Commit:
Fix part of #6550: Write tests for controllers/editor (#6849)

YashJipkate committed:
Fix #6449: Added check for CODEOWNERS to ensure that the most important rules are at the end. (#6560)

alexewu committed:
Fix #6903: added required share link directive to exploration save service to force wrapping (#6908)

* Attempted to fix #6903

* editted CSS in post-publish-modal.html to use flex

* now the elements are aligned in column as seen in the prod version

* Removed previous fix to CSS of modal

* Fix #6903

* added require of sharing-links directive into the exploration save
service

* Fix #6903

* placed require in correct order (alphabetical order)

anubhavsinha98 committed:
Adding solicit_answer_details in the State class (#6810)

* First

* First

* Second

* Third

* Fourth

* Reset

* Added pl json

* Added ask_llearners_for_response

* exp_domain_test completed

* Edited exp_services.py

* editor_test and exp_services_test

* Fixed lint issues

* Updated feconf file

* Ignored package-lock.json

* edited .gitignore

* Made changes in question domain and zip

* Fixing backend tests failure

* Edited question_domain.py

* Fixed backend tests

* Fixed editor_test

* Edited summary for ask_learners_for_response

* Added more tests in exp_services_test

* Added test in state_domain_test.py

* Added more tests

* Added baseline verification

* Edited args docstrings

* Replaced ask_learners_for_response with solicit_answer_details

* Added more changes

* Fixed linting issues

* Sorted order YAML

* Debugging e2e tests

* Debug e2e test

* Removed pre push tests for debugging

* Removed ;

* Replaced earlier e2e test

* Debugging e2e tests

* Added ;

* Made frontend changes

* Added e2e tests

* Corrected a mistake

* Fixed history tab e2e test

* reverted back changes

* Added files

* Fixed branch

* Made some changes

* Changes made

* Added more tests

* Added CANNOT_SOLICIT_ANSWER_DETAILS

* Added changes

* Removed extra new line

* Changes made

* Removed default args

* Removed +

aks681 committed:
fix #6917 (#6918)

DubeySandeep committed:
Add backward linkage of story to the topic it belongs. (#6855)

* Add backward linkage of story to the topic.

* Change random string generator function.

* Fix backend test issue.

* Change belong_to_topic to corresponding_topic_id.

* Fix test issues and address review comments.

* Fix lint issues.

* Adds proper docstring for corresponding_topic_id.

nithusha21 committed:
Add logging for #6702, #6604 (#6833)

* add logging for guppy error, and filter expression error

* fix lint

* fix merge and review comments

Commit:
Fix part of #6550: Write tests for controllers/suggestion (#6822)

* Write tests for controllers/suggestion

* Addressed comments

* Addressed comments

* Addressed comments

* Addressed comments

ankita240796 committed:
Fix #6678: Add test for ensuring usage of angular.mock.inject (#6924)

* Add test for ensuring usage of angular.mock.inject

* Remove extra code, move pattern to bad patterns

* Rename for js and ts

DubeySandeep committed:
Fix #6572: Adds e2e test for written translation project (#6718)

* Init changes.

* Added required e2e tests.

* Fix lint issue.

* Resolved lint issues.

* Fix e2e test issue and add more tests.

* Fix e2e test issue.

* Address review comments.

ankita240796 committed:
Fix #6799: Add restricted globals rule to eslint (#6923)

* Add no-restricted-globals

* Remove globals

* Fix order of dependencies

* Fix test

* Add linebreaks

* Add test

seanlip committed:
Remove unused setup code that is causing breakages on some systems. (#6942)

bansalnitish committed:
Added log for issue #6478 (#6808)

* Add error log for undefined cards

* Add require

* Add missing require

sophiewu6 committed:
Rename PretestEngineService to QuestionPlayerEngineService (#6913)

* Rename PretestEngineService

* Fixed linting errors

* Fixed linting errors

* Fixed linting errors

* Revert package-lock.json

YashJipkate committed:
Move SuggestionModalService to services/ (#6935)

* Move

* Restore CODEOWNERS

* Fix nit

* Fix order

YashJipkate committed:
Clean up work on constants. (#6919)

* Applied stash

* Moved single use constants to page.constants

* Fix e2e by requiring the necessary constants file

* Moved single use constants to services/ where they belong

* Added presubmit check

* Added presubmit check to ensure singular declaration of constants

* Fix lint

* Removed individual declaration of INTERACTION_SPECS

* Test

* Address Review

* Fix tests

* Address review

DubeySandeep committed:
Fix feedback status selector not showing issue. (#6947)

YashJipkate committed:
Migrate standalone controllers in pages to component directives. (#6907)

* Rename and files about page

* Convert to component directive about page

* About Page

Fix lint

Fix lint

* Modify controller

* renaming and files creator dashboard

* Wrap up creator dashboard

* Donate page

* Fix backend tests and about -> about-page, creator-dashboard -> creator-dashboard-page

* Email Dashboard

* Email dashboard Result

* Lint

* Modify

* Wrap up

* Wrap up topic landing

* Lint

* Change .directive -> .controller for about page

* Change .directive -> .controller for creator dashboard and donate

* Fix backend topic editor

* .directive -> .controller email dashboard pages

* Stewards landing page

* Leaner dashboard

* Fix backend

* Fix lint

* Correct path

* Moderator page

* Notifications Dashboard page

* Preferences page

* Practice session page

* Splash page

* RE-added the backend test

* Revert "Practice session page"

This reverts commit 26713bd7cfeb87ce876985fe8e51fe0d72811832.

* Fix e2e in preferences

* Revert "Revert "Practice session page""

This reverts commit bfa7107e91540d12c9b2b1d106da3dc4967d1223.

* removed

* Modify imports

* Modify imports

* Modify imports

* Modify imports

* .directive -> .controller error-page

* Merge conflicts

* Fix merge

DubeySandeep committed:
Fix #6943: Fix code editor interaction issue in exploration player. (#6948)

* Fix code editor interaction issue.

* Address review comments.

varun-tandon committed:
Merge branch 'takeout' of https://github.com/varun-tandon/oppia into takeout

varun-tandon committed:
added nontrivial tests

varun-tandon committed:
resolve linting issues

varun-tandon committed:
more linting

varun-tandon committed:
trailing whitespace

varun-tandon committed:
unused import

varun-tandon committed:
more white space

varun-tandon committed:
new line issues

varun-tandon committed:
blank line

varun-tandon committed:
lint

varun-tandon committed:
lint other file

varun-tandon committed:
remove whitespace
